Adopt more smart pointers in editing code

cdumez opened the pull request:
#### d3c6d562070aae58ed3422466e1cb518c57acb01
<pre>
Adopt more smart pointers in editing code
<a href="https://bugs.webkit.org/show_bug.cgi?id=262452">https://bugs.webkit.org/show_bug.cgi?id=262452</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::identifierForStyleProperty):
(WebCore::fontWeightValueIsBold):
(WebCore::EditingStyle::extractFontSizeDelta):
(WebCore::EditingStyle::conflictsWithInlineStyleOfElement const):
(WebCore::EditingStyle::prepareToApplyAt):
(WebCore::EditingStyle::styleAtSelectionStart):
(WebCore::EditingStyle::inverseTransformColorIfNeeded):
(WebCore::StyleChange::StyleChange):
(WebCore::backgroundColorInEffect):
* Source/WebCore/editing/Editor.cpp:
(WebCore::imageElementFromImageDocument):
(WebCore::Editor::canPaste const):
(WebCore::Editor::shouldInsertText const):
(WebCore::Editor::hasBidiSelection const):
(WebCore::Editor::postTextStateChangeNotificationForCut):
(WebCore::Editor::quoteFragmentForPasting):
(WebCore::Editor::baseWritingDirectionForSelectionStart const):
(WebCore::Editor::setComposition):
(WebCore::Editor::markMisspellingsAfterTypingToWord):
(WebCore::Editor::markMisspellingsOrBadGrammar):
(WebCore::Editor::rangeForPoint):
(WebCore::isFrameInRange):
(WebCore::editableTextListsAtPositionInDescendingOrder):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::verticalScrollDistance):
(WebCore::executeInsertNewline):
(WebCore::valueFormatBlock):
* Source/WebCore/editing/FormatBlockCommand.cpp:
(WebCore::FormatBlockCommand::formatRange):
(WebCore::FormatBlockCommand::elementForFormatBlockCommand):
* Source/WebCore/editing/FormatBlockCommand.h:
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::DragCaretController::caretRectInRootViewCoordinates const):
(WebCore::DragCaretController::editableElementRectInRootViewCoordinates const):
(WebCore::DragCaretController::setCaretPosition):
(WebCore::FrameSelection::setSelectionWithoutUpdatingAppearance):
(WebCore::FrameSelection::setSelection):
(WebCore::FrameSelection::updateSelectionAppearanceNow):
(WebCore::FrameSelection::respondToNodeModification):
(WebCore::adjustPositionForUserSelectAll):
(WebCore::repaintCaretForLocalRect):
(WebCore::FrameSelection::recomputeCaretRect):
(WebCore::CaretBase::invalidateCaretRect):
(WebCore::CaretBase::paintCaret const):
(WebCore::FrameSelection::contains const):
(WebCore::FrameSelection::selectFrameElementInParentIfFullySelected):
(WebCore::FrameSelection::selectAll):
(WebCore::FrameSelection::focusedOrActiveStateChanged):
(WebCore::FrameSelection::updateAppearance):
(WebCore::FrameSelection::selectionBounds):
(WebCore::scanForForm):
(WebCore::FrameSelection::currentForm const):
(WebCore::FrameSelection::setSelectionFromNone):
(WebCore::FrameSelection::dispatchSelectStart):
(WebCore::FrameSelection::setShouldShowBlockCursor):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/InsertIntoTextNodeCommand.cpp:
(WebCore::InsertIntoTextNodeCommand::doApply):
(WebCore::InsertIntoTextNodeCommand::doReapply):
(WebCore::InsertIntoTextNodeCommand::doUnapply):
(WebCore::InsertIntoTextNodeCommand::getNodesInCommand):
* Source/WebCore/editing/InsertIntoTextNodeCommand.h:
(WebCore::InsertIntoTextNodeCommand::protectedNode const):
* Source/WebCore/editing/InsertLineBreakCommand.cpp:
(WebCore::InsertLineBreakCommand::doApply):
* Source/WebCore/editing/InsertNodeBeforeCommand.cpp:
(WebCore::InsertNodeBeforeCommand::doApply):
(WebCore::InsertNodeBeforeCommand::doUnapply):
* Source/WebCore/editing/InsertNodeBeforeCommand.h:
(WebCore::InsertNodeBeforeCommand::protectedInsertChild const):
* Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
(WebCore::highestVisuallyEquivalentDivBelowRoot):
(WebCore::InsertParagraphSeparatorCommand::calculateStyleBeforeInsertion):
(WebCore::InsertParagraphSeparatorCommand::applyStyleAfterInsertion):
(WebCore::InsertParagraphSeparatorCommand::doApply):
* Source/WebCore/editing/InsertParagraphSeparatorCommand.h:
(WebCore::InsertParagraphSeparatorCommand::protectedStyle const):
* Source/WebCore/editing/InsertTextCommand.cpp:
(WebCore::InsertTextCommand::insertTab):
* Source/WebCore/editing/MarkupAccumulator.cpp:
(WebCore::MarkupAccumulator::serializeNodesWithNamespaces):
(WebCore::MarkupAccumulator::entityMaskForText const):
(WebCore::MarkupAccumulator::xmlAttributeSerialization):
* Source/WebCore/editing/MergeIdenticalElementsCommand.cpp:
(WebCore::MergeIdenticalElementsCommand::doApply):
(WebCore::MergeIdenticalElementsCommand::doUnapply):
* Source/WebCore/editing/MergeIdenticalElementsCommand.h:
(WebCore::MergeIdenticalElementsCommand::protectedElement1 const):
(WebCore::MergeIdenticalElementsCommand::protectedElement2 const):
* Source/WebCore/editing/RemoveFormatCommand.cpp:
(WebCore::RemoveFormatCommand::doApply):
* Source/WebCore/editing/RemoveNodeCommand.cpp:
(WebCore::RemoveNodeCommand::doApply):
(WebCore::RemoveNodeCommand::doUnapply):
* Source/WebCore/editing/RemoveNodeCommand.h:
(WebCore::RemoveNodeCommand::protectedNode const):
* Source/WebCore/editing/RemoveNodePreservingChildrenCommand.cpp:
(WebCore::RemoveNodePreservingChildrenCommand::doApply):
* Source/WebCore/editing/RemoveNodePreservingChildrenCommand.h:
(WebCore::RemoveNodePreservingChildrenCommand::portectedNode const):
* Source/WebCore/editing/RenderedPosition.cpp:
(WebCore::rendererFromPosition):
(WebCore::RenderedPosition::positionAtLeftBoundaryOfBiDiRun const):
(WebCore::RenderedPosition::positionAtRightBoundaryOfBiDiRun const):
* Source/WebCore/editing/RenderedPosition.h:
* Source/WebCore/editing/ReplaceNodeWithSpanCommand.cpp:
(WebCore::swapInNodePreservingAttributesAndChildren):
(WebCore::ReplaceNodeWithSpanCommand::doApply):
(WebCore::ReplaceNodeWithSpanCommand::doUnapply):
* Source/WebCore/editing/ReplaceNodeWithSpanCommand.h:
(WebCore::ReplaceNodeWithSpanCommand::protectedSpanElement const):
(WebCore::ReplaceNodeWithSpanCommand::protectedElementToReplace const):
* Source/WebCore/editing/ReplaceRangeWithTextCommand.cpp:
(WebCore::ReplaceRangeWithTextCommand::inputEventDataTransfer const):
* Source/WebCore/editing/ReplaceRangeWithTextCommand.h:
(WebCore::ReplaceRangeWithTextCommand::protectedTextFragment const):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplacementFragment::protectedFragment const):
(WebCore::positionAvoidingPrecedingNodes):
(WebCore::ReplacementFragment::ReplacementFragment):
(WebCore::ReplacementFragment::removeNode):
(WebCore::ReplacementFragment::insertNodeBefore):
(WebCore::ReplacementFragment::insertFragmentForTestRendering):
(WebCore::ReplacementFragment::restoreAndRemoveTestRenderingNodesToFragment):
(WebCore::ReplaceSelectionCommand::ReplaceSelectionCommand):
(WebCore::fragmentNeedsColorTransformed):
(WebCore::ReplaceSelectionCommand::moveNodeOutOfAncestor):
(WebCore::ReplaceSelectionCommand::removeUnrenderedTextNodesAtEnds):
(WebCore::handleStyleSpansBeforeInsertion):
(WebCore::isInlineNodeWithStyle):
(WebCore::ReplaceSelectionCommand::doApply):
(WebCore::ReplaceSelectionCommand::addSpacesForSmartReplace):
(WebCore::ReplaceSelectionCommand::mergeTextNodesAroundPosition):
(WebCore::singleChildList):
(WebCore::deepestSingleChildList):
(WebCore::ReplaceSelectionCommand::insertAsListItems):
(WebCore::ReplaceSelectionCommand::ensureReplacementFragment):
* Source/WebCore/editing/ReplaceSelectionCommand.h:
* Source/WebCore/editing/markup.cpp:
(WebCore::createPageForSanitizingWebContent):
(WebCore::styleFromMatchedRulesAndInlineDecl):
(WebCore::collectElementsToRemoveFromFragment):
(WebCore::replaceChildrenWithFragment):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::protectedNode const):
* Source/WebKitLegacy/mac/WebView/WebHTMLRepresentation.mm:
(-[WebHTMLRepresentation currentForm]):

Canonical link: <a href="https://commits.webkit.org/268709@main">https://commits.webkit.org/268709@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7d5f10083c93a0e75cbf9f220b62c4a08b5a8917

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20433 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20861 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21505 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22331 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19063 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20664 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24118 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21033 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20465 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20652 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20525 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17765 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23183 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17693 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18570 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24856 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18770 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18746 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22796 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19333 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16418 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18547 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4912 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22885 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19156 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/editing/FrameSelection.cpp:
```diff
@@ -431,23 +432,23 @@ bool FrameSelection::setSelectionWithoutUpdatingAppearance(const VisibleSelectio
 
     if (!newSelection.isNone() && !(options & SetSelectionOption::DoNotSetFocus)) {
-        auto* oldFocusedElement = m_document->focusedElement();
+        auto* oldFocusedElement = document->focusedElement();
```
Use RefPtr here as well?

cdumez replied:
It's not being Dereferenced so I wasn't sure but OK.

rniwa commented on Source/WebCore/editing/FrameSelection.cpp:
```diff
@@ -2297,7 +2301,7 @@ void FrameSelection::updateAppearance()
     {
         ScriptDisallowedScope::InMainThread scriptDisallowedScope;
-        auto* view = m_document->renderView();
+        auto* view = document->renderView();
```
Use CheckedPtr here maybe?

rniwa commented on Source/WebCore/editing/FrameSelection.cpp:
```diff
@@ -2321,7 +2325,7 @@ void FrameSelection::updateAppearance()
     // We can get into a state where the selection endpoints map to the same VisiblePosition when a selection is deleted
     // because we don't yet notify the FrameSelection of text removal.
-    if (auto* view = m_document->renderView(); startPos.isNotNull() && endPos.isNotNull() && selection.visibleStart() != selection.visibleEnd()) {
+    if (auto* view = document->renderView(); startPos.isNotNull() && endPos.isNotNull() && selection.visibleStart() != selection.visibleEnd()) {
```
Ditto

cdumez committed:
Adopt more smart pointers in editing code
https://bugs.webkit.org/show_bug.cgi?id=262452

Reviewed by Ryosuke Niwa.

* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::identifierForStyleProperty):
(WebCore::fontWeightValueIsBold):
(WebCore::EditingStyle::extractFontSizeDelta):
(WebCore::EditingStyle::conflictsWithInlineStyleOfElement const):
(WebCore::EditingStyle::prepareToApplyAt):
(WebCore::EditingStyle::styleAtSelectionStart):
(WebCore::EditingStyle::inverseTransformColorIfNeeded):
(WebCore::StyleChange::StyleChange):
(WebCore::backgroundColorInEffect):
* Source/WebCore/editing/Editor.cpp:
(WebCore::imageElementFromImageDocument):
(WebCore::Editor::canPaste const):
(WebCore::Editor::shouldInsertText const):
(WebCore::Editor::hasBidiSelection const):
(WebCore::Editor::postTextStateChangeNotificationForCut):
(WebCore::Editor::quoteFragmentForPasting):
(WebCore::Editor::baseWritingDirectionForSelectionStart const):
(WebCore::Editor::setComposition):
(WebCore::Editor::markMisspellingsAfterTypingToWord):
(WebCore::Editor::markMisspellingsOrBadGrammar):
(WebCore::Editor::rangeForPoint):
(WebCore::isFrameInRange):
(WebCore::editableTextListsAtPositionInDescendingOrder):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::verticalScrollDistance):
(WebCore::executeInsertNewline):
(WebCore::valueFormatBlock):
* Source/WebCore/editing/FormatBlockCommand.cpp:
(WebCore::FormatBlockCommand::formatRange):
(WebCore::FormatBlockCommand::elementForFormatBlockCommand):
* Source/WebCore/editing/FormatBlockCommand.h:
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::DragCaretController::caretRectInRootViewCoordinates const):
(WebCore::DragCaretController::editableElementRectInRootViewCoordinates const):
(WebCore::DragCaretController::setCaretPosition):
(WebCore::FrameSelection::setSelectionWithoutUpdatingAppearance):
(WebCore::FrameSelection::setSelection):
(WebCore::FrameSelection::updateSelectionAppearanceNow):
(WebCore::FrameSelection::respondToNodeModification):
(WebCore::adjustPositionForUserSelectAll):
(WebCore::repaintCaretForLocalRect):
(WebCore::FrameSelection::recomputeCaretRect):
(WebCore::CaretBase::invalidateCaretRect):
(WebCore::CaretBase::paintCaret const):
(WebCore::FrameSelection::contains const):
(WebCore::FrameSelection::selectFrameElementInParentIfFullySelected):
(WebCore::FrameSelection::selectAll):
(WebCore::FrameSelection::focusedOrActiveStateChanged):
(WebCore::FrameSelection::updateAppearance):
(WebCore::FrameSelection::selectionBounds):
(WebCore::scanForForm):
(WebCore::FrameSelection::currentForm const):
(WebCore::FrameSelection::setSelectionFromNone):
(WebCore::FrameSelection::dispatchSelectStart):
(WebCore::FrameSelection::setShouldShowBlockCursor):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/InsertIntoTextNodeCommand.cpp:
(WebCore::InsertIntoTextNodeCommand::doApply):
(WebCore::InsertIntoTextNodeCommand::doReapply):
(WebCore::InsertIntoTextNodeCommand::doUnapply):
(WebCore::InsertIntoTextNodeCommand::getNodesInCommand):
* Source/WebCore/editing/InsertIntoTextNodeCommand.h:
(WebCore::InsertIntoTextNodeCommand::protectedNode const):
* Source/WebCore/editing/InsertLineBreakCommand.cpp:
(WebCore::InsertLineBreakCommand::doApply):
* Source/WebCore/editing/InsertNodeBeforeCommand.cpp:
(WebCore::InsertNodeBeforeCommand::doApply):
(WebCore::InsertNodeBeforeCommand::doUnapply):
* Source/WebCore/editing/InsertNodeBeforeCommand.h:
(WebCore::InsertNodeBeforeCommand::protectedInsertChild const):
* Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
(WebCore::highestVisuallyEquivalentDivBelowRoot):
(WebCore::InsertParagraphSeparatorCommand::calculateStyleBeforeInsertion):
(WebCore::InsertParagraphSeparatorCommand::applyStyleAfterInsertion):
(WebCore::InsertParagraphSeparatorCommand::doApply):
* Source/WebCore/editing/InsertParagraphSeparatorCommand.h:
(WebCore::InsertParagraphSeparatorCommand::protectedStyle const):
* Source/WebCore/editing/InsertTextCommand.cpp:
(WebCore::InsertTextCommand::insertTab):
* Source/WebCore/editing/MarkupAccumulator.cpp:
(WebCore::MarkupAccumulator::serializeNodesWithNamespaces):
(WebCore::MarkupAccumulator::entityMaskForText const):
(WebCore::MarkupAccumulator::xmlAttributeSerialization):
* Source/WebCore/editing/MergeIdenticalElementsCommand.cpp:
(WebCore::MergeIdenticalElementsCommand::doApply):
(WebCore::MergeIdenticalElementsCommand::doUnapply):
* Source/WebCore/editing/MergeIdenticalElementsCommand.h:
(WebCore::MergeIdenticalElementsCommand::protectedElement1 const):
(WebCore::MergeIdenticalElementsCommand::protectedElement2 const):
* Source/WebCore/editing/RemoveFormatCommand.cpp:
(WebCore::RemoveFormatCommand::doApply):
* Source/WebCore/editing/RemoveNodeCommand.cpp:
(WebCore::RemoveNodeCommand::doApply):
(WebCore::RemoveNodeCommand::doUnapply):
* Source/WebCore/editing/RemoveNodeCommand.h:
(WebCore::RemoveNodeCommand::protectedNode const):
* Source/WebCore/editing/RemoveNodePreservingChildrenCommand.cpp:
(WebCore::RemoveNodePreservingChildrenCommand::doApply):
* Source/WebCore/editing/RemoveNodePreservingChildrenCommand.h:
(WebCore::RemoveNodePreservingChildrenCommand::portectedNode const):
* Source/WebCore/editing/RenderedPosition.cpp:
(WebCore::rendererFromPosition):
(WebCore::RenderedPosition::positionAtLeftBoundaryOfBiDiRun const):
(WebCore::RenderedPosition::positionAtRightBoundaryOfBiDiRun const):
* Source/WebCore/editing/RenderedPosition.h:
* Source/WebCore/editing/ReplaceNodeWithSpanCommand.cpp:
(WebCore::swapInNodePreservingAttributesAndChildren):
(WebCore::ReplaceNodeWithSpanCommand::doApply):
(WebCore::ReplaceNodeWithSpanCommand::doUnapply):
* Source/WebCore/editing/ReplaceNodeWithSpanCommand.h:
(WebCore::ReplaceNodeWithSpanCommand::protectedSpanElement const):
(WebCore::ReplaceNodeWithSpanCommand::protectedElementToReplace const):
* Source/WebCore/editing/ReplaceRangeWithTextCommand.cpp:
(WebCore::ReplaceRangeWithTextCommand::inputEventDataTransfer const):
* Source/WebCore/editing/ReplaceRangeWithTextCommand.h:
(WebCore::ReplaceRangeWithTextCommand::protectedTextFragment const):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplacementFragment::protectedFragment const):
(WebCore::positionAvoidingPrecedingNodes):
(WebCore::ReplacementFragment::ReplacementFragment):
(WebCore::ReplacementFragment::removeNode):
(WebCore::ReplacementFragment::insertNodeBefore):
(WebCore::ReplacementFragment::insertFragmentForTestRendering):
(WebCore::ReplacementFragment::restoreAndRemoveTestRenderingNodesToFragment):
(WebCore::ReplaceSelectionCommand::ReplaceSelectionCommand):
(WebCore::fragmentNeedsColorTransformed):
(WebCore::ReplaceSelectionCommand::moveNodeOutOfAncestor):
(WebCore::ReplaceSelectionCommand::removeUnrenderedTextNodesAtEnds):
(WebCore::handleStyleSpansBeforeInsertion):
(WebCore::isInlineNodeWithStyle):
(WebCore::ReplaceSelectionCommand::doApply):
(WebCore::ReplaceSelectionCommand::addSpacesForSmartReplace):
(WebCore::ReplaceSelectionCommand::mergeTextNodesAroundPosition):
(WebCore::singleChildList):
(WebCore::deepestSingleChildList):
(WebCore::ReplaceSelectionCommand::insertAsListItems):
(WebCore::ReplaceSelectionCommand::ensureReplacementFragment):
* Source/WebCore/editing/ReplaceSelectionCommand.h:
* Source/WebCore/editing/markup.cpp:
(WebCore::createPageForSanitizingWebContent):
(WebCore::styleFromMatchedRulesAndInlineDecl):
(WebCore::collectElementsToRemoveFromFragment):
(WebCore::replaceChildrenWithFragment):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::protectedNode const):
* Source/WebKitLegacy/mac/WebView/WebHTMLRepresentation.mm:
(-[WebHTMLRepresentation currentForm]):

Canonical link: https://commits.webkit.org/268709@main
